File: portal/portal/App_Code/DAO/LogOperacaoDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MySql.Data.MySqlClient;
using System.IO;

namespace portal.App_Code.DAO
{
    public class LogOperacaoDAO
    {
        //lendo string de conexão     
        String sc = Properties.Settings.Default.CN;
        String logPath = Properties.Settings.Default.PathErrorLog;

        public LogOperacao carregar(long pCodigo)
        {
            LogOperacao obj = new LogOperacao();
            try
            {
                //cria a conexao com o bd
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Log_Operacao where codigo=" + pCodigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                if (registros.Read())
                {
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Tabela = registros["tabela"].ToString();
                    obj.CodigoChave = Convert.ToInt32(registros["codigo_chave"]);
                    obj.CodigoUsuario = Convert.ToInt32(registros["codigo_usuario"]);
                    obj.TipoOperacao = (LogOperacao.TiposOperacaoes)registros["tipo_operacao"];
                }
                conexao.Close();
            }
            catch (Exception err)
            {
                String log = "Erro=>" + DateTime.Now + err.Message + Environment.NewLine;
                File.AppendAllText(logPath, log);
            }
            return obj;
        }

        public List<LogOperacao> carregarLista(String[] pFiltros, String pOrdena)
        {
            List<LogOperacao> lista = new List<LogOperacao>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Log_Operacao where ";
                for (int i = 0; i < pFiltros.Length; i++)
                {
                    sql = sql + pFiltros[i];
                }
                sql = sql + " order by " + pOrdena;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    LogOperacao obj = new LogOperacao();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Tabela = registros["tabela"].ToString();
                    obj.CodigoChave = Convert.ToInt32(registros["codigo_chave"]);
                    obj.CodigoUsuario = Convert.ToInt32(registros["codigo_usuario"]);
                    obj.TipoOperacao = (LogOperacao.TiposOperacaoes)registros["tipo_operacao"];
                    //obj.Data = Convert.ToInt16(registros["status"]);

                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public static void persistir(LogOperacao obj)
        {
            String sc = Properties.Settings.Default.CN;
            String logPath = Properties.Settings.Default.PathErrorLog;

            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "";
               
                sql = "insert into LogOperacao(tabela, codigo_chave, codigo_usuario, data_operacao, tipo_operacao) value('{0}','{1}','{2}','sysdate',{3})";
               sql = String.Format(sql, obj.Tabela , obj.CodigoChave, obj.CodigoUsuario, obj.TipoOperacao);
               
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }

        

    }
}

File: portal/portal/App_Code/DAO/LogOperacaoDAO.php
<?php
namespace portal\App_Code\DAO {
//CHAMADAS EXTERNAS------------------------------------------------------------------------------------------------------------------------------
    
use portal\App_Code\LogOperacao;     
mysqli_report(MYSQLI_REPORT_STRICT|MYSQLI_REPORT_ERROR);
	
	try{
            require_once("/../LogOperacao.php");        
            require_once("/../../connection.php");        
    }catch(Exception $e){
            require_once (dirname(__file__, 2) . "\LogOperacao.php");
            require_once (dirname(__file__, 3) . "\connection.php");
    }

class LogOperacaoDAO{
//INSERIR----------------------------------------------------------------------------------------------------------------------------------------
    function inserirLogOperacao($log){
        try{
            $con = BD_AbrirConexao();
            $sql = "insert into log_operacao (tabela,codigo_chave,codigo_usuario,data_operacao,tipo_operacao) values (' " .
                 $log->getTabela() . " ',' " .
                 $log->getCodigoChave() . " ',' " .
                 $log->getCodigoUsuario() . " ',' " .
                 $log->getData() . " ',' " .
                 $log->getTipoOperacao() . " ') " ; 
                mysqli_query($con,$sql);
        }catch (Exception $e){
                echo "Erro de inserção: " . $e;
                mysqli_close($con);
                exit;
        }
    echo "Inserção feita com sucesso!";
    mysqli_close($con);	
		}

//ALTERAR---------------------------------------------------------------------------------------------------------------------------------------
    function alterarLogOperacao($log){  
        try{
            $con = BD_AbrirConexao();
            $sql = "update log_operacao set " . 
                "tabela = ' " . $log->getTabela() . " ', " . 
                "codigo_chave = ' " . $log->getCodigoChave() . "', " . 
                "codigo_usuario = ' " . $log->getCodigoUsuario() . "', " .
                "data_operacao = ' " . $log->getData() . " ', " .
                "tipo_operacao = ' " . $log->getTipoOperacao() . " ' " .
                "where codigo = " . $log->getCodigo();
                mysqli_query($con,$sql);
        }catch (Exception $e){
                echo "Erro ao alterar dados: " . $e;
                mysqli_close($con);
                exit;
                }
    echo "Dados alterados com sucesso";
    mysqli_close($con);	
    }

//CARREGAR--------------------------------------------------------------------------------------------------------------------------------------
    function carregarLogOperacao ($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from log_operacao where codigo = " . $codigo;
                $gargantula = mysqli_query($con,$sql);
                $Ulog = new LogOperacao();
                    if($carga = mysqli_fetch_array($gargantula)){
                        $Ulog->setCodigo($carga["codigo"]);
                        $Ulog->setTabela($carga["tabela"]);
                        $Ulog->setCodigoChave($carga["codigo_chave"]);
                        $Ulog->setCodigoUsuario($carga["codigo_usuario"]);
                        $Ulog->setData($carga["data_operacao"]);
                        $Ulog->setTipoOperacao($carga["tipo_operacao"]);
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;
                mysqli_close($con);
                exit;                
            }
        mysqli_close($con);
        return $Ulog;
    }

//CARREGA LISTA DE LOGS------------------------------------------------------------------------------------------------------------------------
    function carregarListaLogs($qtdOpcoes, $ordem){
            $sql = "select * from log_operacao where ";
                    
            for ($i = 0 ; $i <= strlen((array)$qtdOpcoes) ; $i++ ){
                 $sql = $sql . $qtdOpcoes[$i];
				    if($i != $qtdOpcoes[$i] - 1){
                        $sql = $sql . "and ";  
                    } 
            }
            
            $sql = $sql . " order by " . $ordem;
        
            try{
                $con = BD_AbrirConexao();
                $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
                       $listaLog = [];
                            while($carga = mysqli_fetch_array($gargantula)){
                                  $logg = new LogOperacao();
                                  $logg->setLog(($carga["codigo"]),
                                                ($carga["tabela"]),
                                                ($carga["codigo_chave"]),
                                                ($carga["codigo_usuario"]),
                                                ($carga["data_operacao"]),
                                                ($carga["tipo_operacao"]));
                                  $listaLog[] = $logg;
                            }
                        mysqli_close($con);
                        return $listaLog;
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch(Exception $e){
                echo "Erro ao carregar dados: " . $e;
            }
        mysqli_close($con);
    }

//DELETAR----------------------------------------------------------------------------------------------------------------------------------------
    function deletarLogOperacao($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "delete from log_operacao where codigo = " . $codigo; 
                $carga = mysqli_query ($con,$sql);
            }catch(Exception $e){
                echo "Erro de exclusão: " . $e;
                mysqli_close($con);
                exit;                
            }
        echo "Exclusão com sucesso";
        mysqli_close($con);
    }
  }
}
?>